### PR TITLE
259 show timestamps on graphs

### PR DESCRIPTION
**Pull request by jmbrunskill**
**Chart Improvements**

image

- Changed to linear interpolation, easier to see where the actual datapoints are, when we're summarising lots of points it's arguably not as good, but I think it's much easier to read personally.
- Fixed (to some extent) the x-axis disappearing. For some reason it seems to work best with a domain padding and hard coded offset. Still seems slightly different positioning when we have all >0 vs some crossing the 0 line, or all below 0. Can't understand why at this stage!
- Charts are now 24hrs and 48 datapoints are displayed

**Developer Log Generation Tool**
The create sensor logs has been enhanced so it can produce logs for a temperature range, useful for testing the graphs...
image

### CODE REVIEW

**Review by mark-prins — COMMENTED** [review]
Change looks good! haven't tested yet, so I haven't hit 'approve'

**Review by mark-prins — APPROVED** [review]
😎  and 🎉  I think we've got something good here!

One outstanding query was the keyboard height adjustment.. could you explain why that was needed?

File: src/ui/components/settings/SettingsGenerateLogsInputModal.tsx [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          <Row justifyContent="space-between" alignItems="center">
+            <TextInput
+              underlineColorAndroid={COLOUR.SECONDARY}
+              style={{
+                fontSize: FONT.SIZE.L,
+                fontFamily: FONT.FAMILY.REGULAR,
+                color: COLOUR.SECONDARY,
+              }}
+              value={String(numLogs)}
+              keyboardType="numeric"
```
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply src/ui/components/settings/SettingsGenerateLogsInputModal.tsx:86]: Yeah, I guess making is a text input is probably better

File: src/common/constants/Chart.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ export enum CHART {
   NUMBER_OF_TICKS_Y = 5,
   X_KEY = 'timestamp',
   Y_KEY = 'temperature',
-  INTERPOLATION = 'catmullRom',
+  INTERPOLATION = 'linear',
```
[the remark this answers is not in the document]
Comment (jmbrunskill) [reply src/common/constants/Chart.ts:5]: Double checked with Adam. Quote: "the curvey lines are misleading"

**Pull request merged**